libtenzir/builtins/contexts/lookup_table.cpp

```diff
@@ -247,27 +256,55 @@ class ctx final : public virtual context {
       }
       return std::nullopt;
     };
-    for (const auto& value : array.values()) {
+    auto now = time::clock::now();
+    for (auto value : array.values()) {
       if (auto it = context_entries.find(materialize(value));
           it != context_entries.end()) {
-        builder.data(it->second);
+        if ((it->second.create_timeout && it->second.create_timeout < now)
+            || (it->second.update_timeout && it->second.update_timeout < now)) {
+          continue;
+        }
+        if (it->second.update_timeout) {
+          *it.value().update_timeout = now + update_time_;
+        }
+        builder.data(it->second.data);
+        if (it->second.create_timeout) {
+          builder.data(*to<data>(*it->second.create_timeout));
+        } else {
+          builder.data(*to<data>(false));
+        }
+        if (it->second.update_timeout) {
+          builder.data(*to<data>(*it->second.update_timeout));
+        } else {
+          builder.data(*to<data>(false));
+        }
       } else if (auto x = subnet_lookup(value)) {
         builder.data(*x);
+        builder.data(*to<data>(false));
+        builder.data(*to<data>(false));
       } else if (replace and not caf::holds_alternative<caf::none_t>(value)) {
         builder.data(value);
+        builder.data(*to<data>(false));
+        builder.data(*to<data>(false));
```

Comment:
This needs to be undone. There has to be exactly one value or null added to the builder in each iteration over the input series.